media/media.go

```diff
@@ -143,21 +159,34 @@ func (sm svgManager) getFullPath(mediaPath string) string {
 	return filepath.Join(sm.baseDir, mediaPath)
 }
 
-func (sm svgManager) ensureDownloaded(mediaPath string) error {
+func (sm svgManager) ensureDownloaded(mediaPath string, w, h int, c color.Color) (string, error) {
+	// use the colour as a directory to separate different colours of SVG's
+	customizedMediaPath := path.Join(fmt.Sprintf("c%sw%dh%d", colorToHex6(c), w, h), mediaPath)
+
 	// check if we need to download the SVG from the media server
-	_, err := os.Stat(sm.getFullPath(mediaPath))
+	_, err := os.Stat(sm.getFullPath(customizedMediaPath))
 	if errors.Is(err, fs.ErrNotExist) {
 		svg, err := getCachedMediaResource(mediaPath)
 		if err != nil {
-			return err
+			return "", err
 		}
-		err = sm.writeFile(mediaPath, []byte(svg))
+
+		svg = customiseSVG(svg, w, h, c)
+
+		err = sm.writeFile(customizedMediaPath, []byte(svg))
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
-	return nil
+	// return the new media path which includes the colour directory
+	return customizedMediaPath, nil
+}
+
+// customiseSVG wraps the SVG with an outer `svg` tag to ensure that it has the
+// specified width, height and fill attributes.
+func customiseSVG(svg string, w, h int, c color.Color) string {
```

Comment:
This colorization is pretty unique to avatar/head/tail generation, if you wanted to name this specifically. Shrug.

Reply:
I renamed a few things, including this, to make it more clear that it's specific to snake head/tail